superset-core/src/superset_core/api/models.py

```diff
@@ -75,6 +84,83 @@ def backend(self) -> str:
     def data(self) -> dict[str, Any]:
         raise NotImplementedError
 
+    def execute(
+        self,
+        sql: str,
+        options: QueryOptions | None = None,
+    ) -> QueryResult:
```

Comment:
Design consideration: it's not uncommon to have a separate API for only executing queries, vs executing and expecting a result. I'm not necessarily recommending having two separate methods for this, but something to keep in mind.